packages/js/components/src/media-uploader/media-uploader.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Button, DropZone } from '@wordpress/components';
+import { createElement } from 'react';
+import {
+	MediaItem,
+	MediaUpload,
+	uploadMedia as wpUploadMedia,
+	UploadMediaOptions,
+	UploadMediaErrorCode,
+} from '@wordpress/media-utils';
+
+const DEFAULT_ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+type MediaUploaderProps = {
+	allowedMediaTypes?: string[];
+	buttonText?: string;
+	hasDropZone?: boolean;
+	icon?: JSX.Element;
+	label?: string;
+	maxUploadFileSize?: number;
+	MediaUploadComponent?: < T extends boolean = false >(
+		props: MediaUpload.Props< T >
+	) => JSX.Element;
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	onSelect?: ( value: { id: number } & { [ k: string ]: any } ) => void;
+	onError?: ( error: {
+		code: UploadMediaErrorCode;
+		message: string;
+		file: File;
+	} ) => void;
+	onUpload?: ( files: MediaItem[] ) => void;
+	uploadMedia?: ( options: UploadMediaOptions ) => Promise< void >;
+};
+
+export const MediaUploader = ( {
+	allowedMediaTypes = DEFAULT_ALLOWED_MEDIA_TYPES,
+	buttonText = __( 'Choose images', 'woocommerce' ),
+	hasDropZone = true,
+	label = __( 'Drag images here or click to upload', 'woocommerce' ),
+	maxUploadFileSize = 10000000,
+	MediaUploadComponent = MediaUpload,
```

Comment:
Note this component and `uploadMedia` both have defaults which typically work, but are allowed as props for dynamic injection and testing in StoryBook.

This may also prove useful if a user wants to replace the instance of the `MediaUpload` component.